src/ptests/CMakeLists.txt

```
@@ -101,21 +65,16 @@ if (CGNS_ENABLE_FORTRAN AND HAVE_FORTRAN_2003)
     test_unstruc_quad_f90
     test_poly_unstructured_f90
   )
-  set(pcgns_ftest_FILES pcgns_ftest.F90 ${CMAKE_SOURCE_DIR}/src/tests/utilsf.F90)
-  add_executable(pcgns_ftest ${pcgns_ftest_FILES})
-  set(fexample_FILES fexample.F90)
-  add_executable(fexample ${fexample_FILES})
-  set(benchmark_hdf5_f90_FILES benchmark_hdf5_f90.F90)
-  add_executable(benchmark_hdf5_f90 ${benchmark_hdf5_f90_FILES})
-  set(test_mixed_par_ser_FILES test_mixed_par_ser.F90)
-  add_executable(test_mixed_par_ser ${test_mixed_par_ser_FILES})
-  set(test_unstruc_quad_f90_FILES test_unstruc_quad_f90.F90)
-  add_executable(test_unstruc_quad_f90 ${test_unstruc_quad_f90_FILES})
-  set(test_poly_unstructured_f90_FILES test_poly_unstructured_f90.F90 ${CMAKE_SOURCE_DIR}/src/tests/utilsf.F90)
-  add_executable(test_poly_unstructured_f90 ${test_poly_unstructured_f90_FILES})
-  # Linking
+  add_library(test_utilsf OBJECT ${CMAKE_SOURCE_DIR}/src/tests/utilsf.F90)
+
+  macro (ADD_FORTRAN_TEST_EXE file)
+    add_executable (${file} ${file}.F90)
+    target_link_libraries (${file} PRIVATE ${PTESTS_LK_LIBS})
+    target_link_libraries (${file} PRIVATE test_utilsf)
```

```suggestion
    target_link_libraries (${file} PRIVATE ${PTESTS_LK_LIBS} test_utilsf)
```
slight simplification